gunicorn/arbiter.py

```diff
@@ -473,6 +473,8 @@ def manage_workers(self):
         Maintain the number of workers by spawning or killing
         as required.
         """
+        orig_num_workers = self.num_workers
+
         if len(self.WORKERS.keys()) < self.num_workers:
```

Comment:
You're correct, although it makes the code more confusing to check `len(self.WORKERS)` at the top and then `len(workers)` at the bottom.

I suspect we can't change the behavior of `kill_worker` because we need to reap them. We could have a separate dead worker list to reap from, though.